Use swap_tensors path in nn.Module.to for FakeTensor

mikaylagawarecki opened the pull request:
Fixes https://github.com/pytorch/pytorch/issues/148977

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152539

Differential Revision: [D76458023](https://our.internmc.facebook.com/intern/diff/D76458023)

mikaylagawarecki committed:
Use swap_tensors path in nn.Module.to for all subclasses that override __torch_dispatch__

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Use swap_tensors path in nn.Module.to for all subclasses that override __torch_dispatch__"

Fixes https://github.com/pytorch/pytorch/issues/148977




[ghstack-poisoned]

albanD reviewed (approved):
SGTM !

mikaylagawarecki committed:
Update on "Use swap_tensors path in nn.Module.to for all subclasses that override __torch_dispatch__"

Fixes https://github.com/pytorch/pytorch/issues/148977




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge

wdvr commented:
@pytorchmergebot revert -m "failing internal tests - discussed with author" -c ghfirst

mikaylagawarecki committed:
Update on "Use swap_tensors path in nn.Module.to for all subclasses that override __torch_dispatch__"

Fixes https://github.com/pytorch/pytorch/issues/148977




[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D76458023).

mikaylagawarecki committed:
Update on "Use swap_tensors path in nn.Module.to for all subclasses that override __torch_dispatch__"

Fixes https://github.com/pytorch/pytorch/issues/148977


Differential Revision: [D76458023](https://our.internmc.facebook.com/intern/diff/D76458023)

[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D76458023).

albanD reviewed (approved):
PT change looks good!

mikaylagawarecki commented:
@pytorchbot merge
